### PR TITLE
Transition to KiCad 6

### LINKED ISSUE #27: Transition to KiCad 6
Transition to KiCad 6.
- [ ] Update official libraries and utilities submodules to track the `master` branch instead of the `v5` branch
    - [ ] `official-libraries/kicad-symbols`
    - [ ] `official-libraries/kicad-footprints`
    - [ ] `utilities/kicad-library-utils`
- [ ] Update `check-klc-compliance` CI script symbol library identification rules (`*.lib` -> `.kicad_sym`): https://github.com/apcountryman/prototype-kicad-library-ci/blob/e87c0822ec90e63fefd866abd258669db11a1e42/ci/check-klc-compliance#L56
- [ ] Update test library/project KiCad files
    - [ ] `fp-lib-table`
    - [ ] `sym-lib-table`
    - [ ] `test.dcm`
    - [ ] `test.lib` -> `test.kicad_sym`
    - [ ] `test.pretty/10_uMAX.kicad_mod`
    - [ ] `test.pro`

### PR DESCRIPTION
**Pull request by jaylamb**
Transition repository to support KiCad version 6

Resolves #27.

This pull request:
- [ ] Implements a bug fix
- [X] Implements an enhancement to an existing feature
- [ ] Implements a new feature
- [ ] Performs a refactoring

Please mark the pull request as "Ready for review" and request a review when the pull
request is ready for a review.
If changes are requested, please discuss and/or address the review findings before
requesting a new review.

@apcountryman

### CODE REVIEW

**Comment by apcountryman** [comment]
I have not set up KiCad 6 on my computer yet so it may take a few days to get my review done.

File: sym-lib-table [hunk 1]
```diff
@@ -1,3 +1,3 @@
 (sym_lib_table
-  (lib (name test)(type Legacy)(uri ${KIPRJMOD}/test.lib)(options "")(descr ""))
+  (lib (name "test")(type "KiCad")(uri "${KIPRJMOD}/test.kicad_sym")(options "")(descr ""))
```
Comment (apcountryman) [inline sym-lib-table:2]: Did KiCad generate the quotes for the name, type, and uri fields or were they added manually?

`fp-lib-table` doesn't use quotes for these fields. Should it be updated to use quotes for consistency with `sym-lib-table`?

File: test.kicad_prl [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    "zone_display_mode": 0
+  },
+  "meta": {
+    "filename": "test.kicad_prl",
+    "version": 3
+  },
+  "project": {
+    "files": []
+  }
+}
```
Comment (apcountryman) [inline test.kicad_prl]: According to https://docs.kicad.org/6.0/en/kicad/kicad.html#kicad_files_and_folders, this file contains the last used settings for the project and probably shouldn't be under version control, in which case it should be removed and a `.gitignore` rule should be added.

File: .gitignore [hunk 3]
```diff
@@ -5,3 +5,6 @@
 
 # ignore the KiCad footprint cache
 fp-info-cache
+
+# ignore project backups
+*-backups*
```
Comment (apcountryman) [inline .gitignore]: According to https://docs.kicad.org/6.0/en/getting_started_in_kicad/getting_started_in_kicad.html#tutorial_part_1_project and my testing, it looks like project backups are placed in a directory that uses a naming scheme of `<project name>-backups`. Based on this, the following would be a more appropriate ignore rule:
```suggestion
*-backups/
```

**Review by apcountryman — CHANGES_REQUESTED** [review]
The following submodules are significantly out-of-date:
- `utilities/kicad-library-utils`
- `official-libraries/kicad-footprints`
- `official-libraries/kicad-symbols`

Please update these submodules to more recent commits.

File: sym-lib-table (hunk 1 above)
Comment (jaylamb) [reply sym-lib-table:2]: The quotes were automatically generated when I migrated the symbols library over within KiCad 6.

As a test, I removed the file and had KiCad regenerate it, by re-adding the library from `Manage Symbol Libraries` within KiCad. The newly generated file was identical to the previous one, with the quotes in the same places.

**Comment by apcountryman** [comment]
I just updated my KiCad mirrors and will re-run the failed checks.

**Review by apcountryman — APPROVED** [review]
LGTM

**Pull request merged**